helm-v3/006-repositories.md

```diff
@@ -1,40 +1,153 @@
 # Repositories
 
-The following changes will be made to the repository subsystem:
+The following major changes will be made as it relates to working with
+chart repositories in Helm 3:
 
+- A new Capabilities API will be introduced that will allow users to
+opt-in to certain repository functionality
+- A new `helm login` command will be added for authenticating against a
+repository provider
+- A new `helm push` command will be added for pushing charts to a
+repository
 - The `helm serve` command will be removed
-- A new `helm push` verb will be added for pushing charts to a repository
-- The index file will be JSON instead of YAML
-
-> This document is in-progress and needs input from Chart Museum.
-
-## Pushing Charts To A Repository
-
-Several registries have emerged with support for Helm charts including
-ChartMuseum and Quay (via Application Registries) along with extras built on
-top of other systems, such as object storage (e.g., see the [S3
-Plugin](https://github.com/hypnoglow/helm-s3)). These registries are asking for
-and could benefit from a capability to push packages from the Helm client.
-
-To support registries there will be a `helm push` command that works in a
-similar manner to `helm fetch`. It will support different systems via the
-URI/URL scheme and provide a default implementation for HTTP(S). Other systems,
-such as S3, can implement an uploader via a plugin for schemes Helm does not
-provide an implementation for.
-
-> [name=Matt Farina] There very well may be additional work here around
-> registry search and pluggable or otherwise extendable authenication
-> mechanisms.
-> 
-> [name=Adnan Abdulhussein] What would the default HTTP(S) implementation be?
-> [name=Matt Fisher] Adnan, would you mind clarifying your comment a little?
-> 
-> [name=Adnan Abdulhussein] Sorry, the proposal mentions there will be a
-> default implemention for `helm push` to a regular HTTP(S) chart repository.
-> It's unclear to me how this would actually work.
-> 
-> [name=Adnan Abdulhussein] Sounds like this is something that will be fleshed
-> out more in Josh's proposal, so I'll wait for that :)
+- A repository v2 specification will be defined, and the index file will be
+JSON (`index.json` vs `index.yaml`)
```

Comment:
can the index file support pagination? General a huge index file is not efficient from server side.

Reply:
you might be interested in https://github.com/helm/community/blob/master/proposals/distributed-search.md which is a separate proposal

Reply:
whether pulling a single or multiple charts from a scope, it would be great if we assumed there are far more responses then a client would actually want. Having paging by default in any result would provide the scaling we need.